models/cart.model.js

```diff
@@ -0,0 +1,18 @@
+const mongoose = require("mongoose");
+const { Schema } = require("mongoose");
+const { User } = require("../models/user.model");
+const { Product } = require("../models/product.model");
+const CartSchema = new mongoose.Schema(
```

Comment:
This schema creates a new instance for every product in the cart collection. for example.
```
//product 1
productId: "product1",
quantity: "1",
user: "user1"

//product 2
productId: "product2",
quantity: "8",
user: "user1"
````
This works too, but you can have a single instance like this
```
userId: "user1"
cart: [ { product: "product1", quantity: 1}, { product: "product2", quantity: 8} ]  // type array
```
`schema` for above: 
```
const CartSchema = new Schema({
  user: {
    type: Schema.Types.ObjectId, 
    ref: 'User',
    required: true
  },
  cartItems: [
    {
      product: { type: Schema.Types.ObjectId, ref: "Product", required: true },
      quantity: { type: Number, default: 1 }
    }
  ]
})
````